### ISSUE TITLE
legend symbols misplaced when more than one fill exists

### ISSUE THREAD

**Issue by lindsayplatt** [USGS-R/gsplot#302]
legend places symbols in the wrong location when more than one rectangle is added

``` r
#two rectangles
gs <- gsplot() %>% 
  rect(xleft = c(1,2), xright = c(2,4),
       ybottom = c(1,1), ytop = c(2,2),
       col = c("red", "green"), legend.name=c("one", "two")) %>%
  points(5:7, 5:7, pch=24, col="black", bg="grey", legend.name="field visits") %>% 
  legend()
gs
```

![image](https://cloud.githubusercontent.com/assets/13220910/12486430/57647f4a-c028-11e5-86dd-d5ac248a66a1.png)

Compared to:

``` r
gs <- gsplot() %>% 
    rect(xleft = 1, xright = 2,
         ybottom = 1, ytop = 2,
         col = "green", legend.name="one") %>%
    points(5:7, 5:7, pch=24, col="black", bg="grey", legend.name="field visits") %>% 
    legend()
gs
```

![image](https://cloud.githubusercontent.com/assets/13220910/12486449/6c2aaf62-c028-11e5-989e-696201bec92c.png)

**Comment by lindsayplatt** [comment]
My suggested solution to this is to pass in each legend item separately to the `set_legend_args` function. So, when `length(legend.name) > 1`, you should separate the parameters into multiple `paramsAll`'s and pass them through the legend args logic. See my PR that does this #305

**Comment by lindsayplatt** [comment]
This issue has been fixed by #305, but is not the cleanest of solutions. Will open a new issue about making it cleaner
